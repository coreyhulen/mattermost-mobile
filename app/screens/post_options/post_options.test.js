// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import PostOptions from './post_options';

jest.mock('react-intl');

describe('PostOptions', () => {
    const navigator = {
        showModal: jest.fn(),
        dismissModal: jest.fn(),
    };

    const actions = {
        addReaction: jest.fn(),
        deletePost: jest.fn(),
        flagPost: jest.fn(),
        pinPost: jest.fn(),
        removePost: jest.fn(),
        unflagPost: jest.fn(),
        unpinPost: jest.fn(),
    };

    const post = {
        id: 'post_id',
        message: 'message',
    };

    const baseProps = {
        actions,
        canAddReaction: true,
        canDelete: true,
        canPin: true,
        canEdit: true,
        canEditUntil: -1,
        channelIsReadOnly: false,
        currentTeamUrl: 'http://localhost:8065/team-name',
        deviceHeight: 600,
        hasBeenDeleted: false,
        isFlagged: false,
        isMyPost: true,
        isSystemMessage: false,
        managedConfig: {},
        navigator,
        post,
        showAddReaction: true,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot, showing all possible options', () => {
        const wrapper = shallow(
            <PostOptions {...baseProps}/>,
            {context: {intl: {formatMessage: jest.fn((format) => format.defaultMessage)}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, showing Delete option only for system message to user who has permission to delete', () => {
        const wrapper = shallow(
            <PostOptions
                {...baseProps}
                isSystemMessage={true}
            />,
            {context: {intl: {formatMessage: jest.fn((format) => format.defaultMessage)}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot, no option for system message to user who doesn\'t have the permission to delete', () => {
        const wrapper = shallow(
            <PostOptions
                {...baseProps}
                isSystemMessage={true}
                canDelete={false}
            />,
            {context: {intl: {formatMessage: jest.fn((format) => format.defaultMessage)}}},
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
