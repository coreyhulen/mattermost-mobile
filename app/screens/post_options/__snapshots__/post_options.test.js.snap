// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostOptions should match snapshot, no option for system message to user who doesn't have the permission to delete 1`] = `null`;

exports[`PostOptions should match snapshot, showing Delete option only for system message to user who has permission to delete 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(SlideUpPanel)
    allowStayMiddle={false}
    initialPosition={124}
    marginFromTop={476}
    onRequestClose={[Function]}
  >
    <PostOption
      destructive={true}
      icon="trash"
      onPress={[Function]}
      text="Delete"
    />
  </Connect(SlideUpPanel)>
</View>
`;

exports[`PostOptions should match snapshot, showing all possible options 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(SlideUpPanel)
    allowStayMiddle={false}
    initialPosition={424}
    marginFromTop={176}
    onRequestClose={[Function]}
  >
    <PostOption
      icon="edit"
      onPress={[Function]}
      text="Edit"
    />
    <PostOption
      icon="flag"
      onPress={[Function]}
      text="Flag"
    />
    <PostOption
      icon="pin"
      onPress={[Function]}
      text="Pin to Channel"
    />
    <PostOption
      icon="emoji"
      onPress={[Function]}
      text="Add Reaction"
    />
    <PostOption
      icon="link"
      onPress={[Function]}
      text="Copy Permalink"
    />
    <PostOption
      icon="copy"
      onPress={[Function]}
      text="Copy Text"
    />
    <PostOption
      destructive={true}
      icon="trash"
      onPress={[Function]}
      text="Delete"
    />
  </Connect(SlideUpPanel)>
</View>
`;
